File: python/80_删除有序数组中的重复项_II_medium.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        n, j = len(nums), 2
        if n <= 2:
            return n
        for i in range(2, n):
            if nums[i] != nums[j-2]:
                nums[j] = nums[i]
                j += 1
        return j

File: cpp/80_删除有序数组中的重复项_II_medium.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int n = nums.size();
        if (n<=2) return n;
        int j = 2;
        for (int i=2; i<n; i++)
            if (nums[i] != nums[j-2]) {
                nums[j] = nums[i];
                j++;
            }
        return j;
    }
};